### ISSUE TITLE
Caching prevents warnings from configuration within source code

### ISSUE THREAD

**Issue by tnir** [rubocop/rubocop#10822]
## Expected behavior

Even when [caching](https://docs.rubocop.org/rubocop/usage/caching.html) is not disabled, warnings from the configuration in source code should be emitted like ones from the configuration in `.rubocop.yml`:
```
# 1st time
$ rubocop . --cache true
.rubocop.yml: NotExist/ClassVars has the wrong namespace - should be Style
Inspecting 1 file
rubocop_cache.rb: NotExist/ClassVars has the wrong namespace - should be Style
.

1 file inspected, no offenses detected
# 2nd time
$ rubocop . --cache true
.rubocop.yml: NotExist/ClassVars has the wrong namespace - should be Style
Inspecting 1 file
rubocop_cache.rb: NotExist/ClassVars has the wrong namespace - should be Style
.

1 file inspected, no offenses detected
```

- Context: https://github.com/rubygems/rubygems/pull/5730

## Actual behavior

When the file cached, warnings from the configuration in source code are not emitted while ones from the configuration in `.rubocop.yml` are emitted:
```
# 1st time
$ rubocop . --cache true
.rubocop.yml: NotExist/ClassVars has the wrong namespace - should be Style
Inspecting 1 file
rubocop_cache.rb: NotExist/ClassVars has the wrong namespace - should be Style
.

1 file inspected, no offenses detected
# 2nd time
$ rubocop . --cache true
.rubocop.yml: NotExist/ClassVars has the wrong namespace - should be Style
Inspecting 1 file
.

1 file inspected, no offenses detected
```

Compare with the results on cache disabled:
```
$ rubocop . --cache false
.rubocop.yml: NotExist/ClassVars has the wrong namespace - should be Style
Inspecting 1 file
rubocop_cache.rb: NotExist/ClassVars has the wrong namespace - should be Style
.

1 file inspected, no offenses detected
$ rubocop . --cache false
.rubocop.yml: NotExist/ClassVars has the wrong namespace - should be Style
Inspecting 1 file
rubocop_cache.rb: NotExist/ClassVars has the wrong namespace - should be Style
.

1 file inspected, no offenses detected
```

## Steps to reproduce the problem

```yml
$ cat .rubocop.yml 
NotExist/ClassVars:
  Enabled: true

AllCops:
  DisabledByDefault: true
```

```ruby
$ cat rubocop_cache.rb 
class RubocopCache
  @@hello = nil # rubocop:disable NotExist/ClassVars
end
```

## RuboCop version

```
$ rubocop -V
1.31.2 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 2.7.6 x86_64-linux)
```

**Comment by jonas054** [comment]
I agree. Just like you'd expect the same offenses to be reported each time, you'd also expect to get the same warnings.

Since the cache is an offense cache that doesn't support storing warnings currently, I think the easiest solution would be to not store anything in the cache for files that get warnings. I'll try to work out a solution.

**Comment by dgollahon-plaid** [comment]
This should be re-opened--AFAICT it has not been fixed and it's been a problem for years.

Bad bot.
